### PR TITLE
Feature/feature crud picture

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ src/main/java/org/pesmypetcare/webservice/firebaseservice/FirebaseFactory.java  1
+ src/main/java/org/pesmypetcare/webservice/dao/UserDaoImpl.java  1
+ src/main/java/org/pesmypetcare/webservice/error/RestExceptionHandler.java  25
+ src/main/java/org/pesmypetcare/webservice/error/DatabaseAccessException.java  25
+ src/main/java/org/pesmypetcare/webservice/error/ErrorBody.java  50
         

Complexity increasing per file
==============================
- src/test/java/org/pesmypetcare/webservice/service/StorageServiceTest.java  1
- src/main/java/org/pesmypetcare/webservice/service/StorageServiceImpl.java  1
- src/test/java/org/pesmypetcare/webservice/dao/PetDaoTest.java  2
- src/main/java/org/pesmypetcare/webservice/form/StorageForm.java  1
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/StorageRestController.java  1
- src/main/java/org/pesmypetcare/webservice/dao/StorageDaoImpl.java  2
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/StorageRestControllerTest.java  1
- src/test/java/org/pesmypetcare/webservice/dao/StorageDaoTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5305936&bid=17365499)

File: src/main/java/org/pesmypetcare/webservice/dao/PetDaoImpl.java [hunk 1]
```diff
@@ -26,27 +26,45 @@
...
+        storageDao = new StorageDaoImpl(this);
 
         PETS_KEY = "pets";
         DELFAIL_KEY = "deletion-failed";
         PET_DOES_NOT_EXIST_EXC = "The pet does not exist";
         INVALID_PET_EXC = "invalid-pet";
     }
 
+    /**
+     * Gets the storage dao
```
Comment (xavier-campos-diaz) [inline src/main/java/org/pesmypetcare/webservice/dao/PetDaoImpl.java]: Missing a period at the end of the comment

**Review by xavier-campos-diaz — COMMENTED** [review]
You need to check some issues 

DlMrh3zVAAEuayd

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/org/pesmypetcare/webservice/controller/appmanager/StorageRestControllerTest.java  1
- src/main/java/org/pesmypetcare/webservice/dao/StorageDaoImpl.java  2
- src/main/java/org/pesmypetcare/webservice/service/StorageServiceImpl.java  1
- src/main/java/org/pesmypetcare/webservice/form/StorageForm.java  1
- src/test/java/org/pesmypetcare/webservice/dao/StorageDaoTest.java  1
- src/test/java/org/pesmypetcare/webservice/dao/PetDaoTest.java  2
- src/test/java/org/pesmypetcare/webservice/service/StorageServiceTest.java  1
- src/main/java/org/pesmypetcare/webservice/controller/appmanager/StorageRestController.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5305936&bid=17365499)

**Review by xavier-campos-diaz — APPROVED** [review]

File: src/main/java/org/pesmypetcare/webservice/dao/PetDaoImpl.java (hunk 1 above)
Comment (santidrj) [reply src/main/java/org/pesmypetcare/webservice/dao/PetDaoImpl.java]: Not entirely my fault, Codacy didn't check it as an issue either.

**Review by marclll — APPROVED** [review]
Well done boy

**Pull request merged**